pkg/odo/cli/component/create.go

```diff
@@ -319,7 +319,11 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
 					supDevfileCatalogList = append(supDevfileCatalogList, devfileComponent)
 				}
 			}
-			componentType = ui.SelectDevfileComponentType(supDevfileCatalogList)
+
```

Comment:
as [1] is the first check, if a `<component-type` not present and `devfile.yaml` is present, it is default going into interactive, IIUC it should create from existing `devfile.yaml`
[1]
```
if len(args) == 0 {
  co.interactive = true
	}
```

Reply:
Yes, it should create the component using the existing `devfile.yaml`, but user is still able to specify the `<component name>` via interactive mode, that's why it falls down to interactive mode